panther_gazebo/src/led_strip.cpp

```diff
@@ -0,0 +1,279 @@
+// Copyright 2024 Husarion sp. z o.o.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "panther_gazebo/led_strip.hpp"
+
+#include <stddef.h>
+
+#include <gz/plugin/Register.hh>
+#include <gz/sim/Model.hh>
+#include <gz/sim/components/LightCmd.hh>
+#include <gz/sim/components/Name.hh>
+
+#include <gz/msgs/color.pb.h>
+#include <gz/msgs/marker.pb.h>
+namespace panther_gazebo
+{
+void LEDStrip::Configure(
+  const gz::sim::Entity & entity, const std::shared_ptr<const sdf::Element> & sdf,
+  gz::sim::EntityComponentManager & ecm, gz::sim::EventManager & /*eventMgr*/)
+{
+  const auto model = gz::gazebo::Model(entity);
+  if (!model.Valid(ecm)) {
+    throw std::runtime_error(
+      "Error: Failed to initialize because [" + model.Name(ecm) +
+      "] (Entity=" + std::to_string(entity) +
+      ") is not a model. Please make sure that LEDStrip is attached to a valid model.");
+  }
+
+  ParseParameters(sdf);
+  light_cmd_ = SetupLightCmd(ecm);
+
+  node_.Subscribe(ns_ + "/" + image_topic_, &LEDStrip::ImageCallback, this);
+}
+
+void LEDStrip::PreUpdate(const gz::sim::UpdateInfo & info, gz::sim::EntityComponentManager & ecm)
+{
+  auto current_time = info.simTime;
+  auto period = std::chrono::milliseconds(static_cast<int>(1000 / frequency_));
+
+  if (new_image_available_ && current_time - last_update_time_ >= period) {
```

Comment:
Let's return function when statement is not met.  
```suggestion
  if (!new_image_available_ || (current_time - last_update_time_)  < period) {
  return;
  }
```